may21readme-update.md

```diff
@@ -0,0 +1,309 @@
+```
+#### Installing truffle
+Simply run:
+```
+npm install -g truffle
+```
+#### Installing mongoDB
+Please remember to make sure that you are running `glibc 2.23`. To make sure, run `ldd --version` in terminal to check.
+
+Follow the steps from the [following manual.](https://hevodata.com/blog/install-mongodb-on-ubuntu/).
+
+
+#### Installing the IPFS daemon
+We use IPFS (more information about this alternative to http found [here](https://ipfs.io/#why)) because ...
+
+Update your packages and make sure that you have a version of Go beyond 1.7.0
+```
+sudo apt-get update
+sudo apt-get install golang-go -y
+```
+
+Get the tarball from the web, then untar it.
+```
+wget https://dist.ipfs.io/go-ipfs/v0.4.10/go-ipfs_v0.4.10_linux-386.tar.gz
+tar xvfz go-ipfs_v0.4.10_linux-386.tar.gz
+```
+
+Move the `ipfs` binary in your local path.
+```
+sudo mv go-ipfs/ipfs /usr/local/bin/ipfs
+```
+
+To test that it is working, type
+```
+ipfs help
+```
+This will yield something like:
+```
+USAGE:
+
+    ipfs - Global p2p merkle-dag filesystem.
+...
+```
+#### Installing MetaMask
+
+Go to [this](https://metamask.io/) site in your Google Chrome browser.
+
+
+### Installing Distribute Protocol
+Note that all of these steps _must_ be carried out before sending any transactions on distribute.
+
+*Step 1:*    
+Go to Distribute Protocol's frontend. Navigate to its [`contracts` folder](https://github.com/distribute-protocol/distribute-contracts.git) and clone it. To do this, navigate to a suitable directory in Terminal and type:
+```
+git clone https://github.com/distribute-protocol/distribute-contracts.git
+cd contracts
+```
+*Step 2:*    
+Run the nodeJS modules and dependencies from the package.json file.
+```
+yarn
+yarn start
+```
+*Step 3*:   
+Open a new terminal window. Type
+```
+ganache-cli
+```
+This will deploy a local blockchain with 10 accounts, each preloaded with 100 ether.
+
+*Step 4:*   
+Open and sign in to MetaMask in your Google Chrome browser.
+From the dropdown menu at the top, select the `Localhost 8545` network. Click the three dots and select "Copy Address to Clipboard".
+
+Remember to reset this account everytime you run `ganache-cli`. To reset the account, confirm that you are on Localhost 8545, then navigate to Settings from the three bars on the top right. Scroll down and select "Reset Account". This forces MetaMask to believe that this network ID is a clean network in a fresh state (ordinary users of MetaMask should never had to do this!).
+
+*Step 5:*   
+In a new Terminal window, navigate to the `contracts` folder you cloned from github in Step 1. Deploy it to the local blockchain running in ganache by typing:
+```
+truffle migrate
+```
+Then enter the Truffle console.
+```
+truffle console
+```
+Send ether from your local blockchain to your account in MetaMask by typing the following in `truffle console`:
+```
+web3.eth.sendTransaction({to:"MetaMask address", from: web3.eth.accounts["i"], value: web3.toWei(90, 'ether')})
```

Comment:
it's convention to write something like ({to: '<MetaMask address>', from: web3.eth.accounts[<i>]... to avoid confusion between parameters that must be input as strings (the address) and those that should not (the index)